lib/sisimai/bite/email/gsuite.rb

```diff
@@ -0,0 +1,273 @@
+        :networkerror => %r/DNS type .+ lookup of .+ responded with code NXDOMAIN/,
+      }
+      Indicators = Sisimai::Bite::Email.INDICATORS
+
+      def description; return 'G Suite: https://gsuite.google.com'; end
+      def smtpagent;   return Sisimai::Bite.smtpagent(self); end
+      def headerlist;  return ['X-Gm-Message-State']; end
+      def pattern;     return Re0; end
+
+      # Parse bounce messages from G Suite (Transfer from G Suite to a destinaion host)
+      # @param         [Hash] mhead       Message headers of a bounce email
+      # @options mhead [String] from      From header
+      # @options mhead [String] date      Date header
+      # @options mhead [String] subject   Subject header
+      # @options mhead [Array]  received  Received headers
+      # @options mhead [String] others    Other required headers
+      # @param         [String] mbody     Message body of a bounce email
+      # @return        [Hash, Nil]        Bounce data list and message/rfc822
+      #                                   part or nil if it failed to parse or
+      #                                   the arguments are missing
+      def scan(mhead, mbody)
+        return nil unless mhead
+        return nil unless mbody
+
+        return nil unless mhead['from']    =~ Re0[:from]
+        return nil unless mhead['subject'] =~ Re0[:subject]
+        return nil unless mhead['x-gm-message-state']
+
+        require 'sisimai/address'
+        dscontents = [Sisimai::Bite.DELIVERYSTATUS]
+        hasdivided = mbody.split("\n")
+        rfc822list = []     # (Array) Each line in message/rfc822 part string
+        blanklines = 0      # (Integer) The number of blank lines
+        readcursor = 0      # (Integer) Points the current cursor position
+        recipients = 0      # (Integer) The number of 'Final-Recipient' header
+        endoferror = true   # (Integer) Flag for a blank line after error messages
+        anotherset = {}     # (Hash) Another error information
+        emptylines = 0      # (Integer) The number of empty lines
+        connvalues = 0      # (Integer) Flag, 1 if all the value of $connheader have been set
+        connheader = {
+          'date'  => '',    # The value of Arrival-Date header
+          'lhost' => '',    # The value of Reporting-MTA header
+        }
+        v = nil
+
+        hasdivided.each do |e|
+          if readcursor.zero?
+            # Beginning of the bounce message or delivery status part
+            if e =~ Re1[:begin]
+              readcursor |= Indicators[:deliverystatus]
+            end
+          end
+
+          if readcursor & Indicators[:'message-rfc822'] == 0
+            # Beginning of the original message part
+            if e =~ Re1[:rfc822]
+              readcursor |= Indicators[:'message-rfc822']
+              next
+            end
+          end
+
+          if readcursor & Indicators[:'message-rfc822'] > 0
+            # After "message/rfc822"
```

Comment:
__[rubocop]__
* Style/NumericPredicate: Use `(readcursor & Indicators[:'message-rfc822']).zero?` instead of `readcursor & Indicators[:'message-rfc822'] == 0`. [:ref](https://github.com/bbatsov/ruby-style-guide#predicate-methods)

<!-- Comment by [SideCI](https://sideci.com) -->